include/zephyr/sys/time_units.h

```diff
@@ -138,6 +135,16 @@ static TIME_CONSTEXPR inline int sys_clock_hw_cycles_per_sec(void)
 	 (__round_up) ? ((__from_hz) / (__to_hz)) - 1 :			\
 	 0)
 
+/* Clang emits a divide-by-zero warning even though the int_div macro
+ * results are only used when the divisor will not be zero. Work
+ * around this by substituting 1 to make the compiler happy.
+ */
+#ifdef __clang__
+#define z_tmcvt_divisor(a, b) ((a) / (b) ?: 1)
```

Comment:
At what level of the language standard did that ternary syntax appear?  I worry because we have `__clang__` compilers in the stable (c.f. Cadence's xt-clang) which are comparatively old.  Might be safer to write that out long hand?  Or not, if we're sure this is supported.

Reply:
Also grade school math pedantry: the operation of this macro should probably be named "quotient", or just "divide", the "divisor" is the `a` argument.

Reply:
Yeah, the name actually refers to the division that this value is used as a part of -- which is in the `z_tmcvt_int_div_*` macros. In that context, it looks like I guessed right in using 'divisor' (I admit that I just checked :-)